src/pages/workspaces/workspace/analysis/runtime-common.js

```diff
@@ -92,8 +92,11 @@ export const RuntimeStatusMonitor = ({ runtime, onRuntimeStoppedRunning = _.noop
 }
 
 export const AuthenticatedCookieSetter = () => {
-  const { registrationStatus } = useStore(authStore)
-  return registrationStatus === userStatus.registeredWithTos && getLocalPref(cookiesAcceptedKey) !== false ? h(PeriodicCookieSetter) : null
+  const { termsOfService } = useStore(authStore)
+  const cookiesAccepted = getLocalPref(cookiesAcceptedKey) !== false
+  const userCanUseTerra = termsOfService.userCanUseTerra
+
+  return userCanUseTerra && cookiesAccepted ? h(PeriodicCookieSetter) : null
```

Comment:
This reads a lot better, thanks! I'll likely just wait for this PR at this point, since the tests are being finicky.